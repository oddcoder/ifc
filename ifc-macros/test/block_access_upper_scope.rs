use ifc_macros::ifc_block;
fn main() {
    ifc_block!{
        let x = 5;
        {
            // this works best as a failing test
            // since we need to make sure that x
            // is not typed as None and instead read from
            // higher scope which is typed as High
            #[IFC[Low]]
            let _y = x;
        }
    }
}