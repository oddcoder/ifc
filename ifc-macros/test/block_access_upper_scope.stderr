error: Cannot assign high expression to low variable
  --> $DIR/block_access_upper_scope.rs:11:13
   |
11 |             let _y = x;
   |             ^^^^^^^^^^^
   |
help: Low Variable
  --> $DIR/block_access_upper_scope.rs:11:17
   |
11 |             let _y = x;
   |                 ^^
help: High Expression
  --> $DIR/block_access_upper_scope.rs:11:22
   |
11 |             let _y = x;
   |                      ^
